examples/create-program-reentrance/src/wasm.rs

```diff
@@ -0,0 +1,71 @@
+// This file is part of Gear.
+
+// Copyright (C) 2023-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use gstd::{debug, msg, prog, ActorId, CodeId};
+
+static mut DST_PROGRAM_ID: ActorId = ActorId::new([0u8; 32]);
+
+type HandleParams = ([u8; 32], u128);
+
+#[no_mangle]
+extern "C" fn init() {
+    let program_id: [u8; 32] = msg::load().expect("internal error: invalid payload");
+    debug!(
+        "[CREATE_PROGRAM_REENTRANCE::init()] DST_PROGRAM_ID: 0x{}",
+        hex::encode(program_id)
+    );
+
+    unsafe { DST_PROGRAM_ID = program_id.into() };
+}
+
+#[gstd::async_main]
+async fn main() {
+    let (code_hash, amount): HandleParams = msg::load().expect("internal error: invalid payload");
+    debug!(
+        "[CREATE_PROGRAM_REENTRANCE::handle] code_hash: 0x{}, amount: {}",
+        hex::encode(code_hash),
+        amount
+    );
+
+    let code_id: CodeId = code_hash.into();
+    let dest_program_id = unsafe { DST_PROGRAM_ID };
+
+    // Call gr_create_program syscall
+    let (_msg_id, program_id) =
```

Comment:
seems you could just create program, assert balance within the exec
than call it again to just share the info about balance and the balance should be the same if program wasn't created

Reply:
sending out a message and waiting for reply is crucial to let the `value_counter` to be reset between executions when a message goes through the wait-wake cycle. This should be a part of the test.
Or not?

Reply:
yeah, that's ok, it was just an option to simplify the test
unnecessary to follow

the aim was to assert prog balance directly inside wasm program